modules/broker-cli-client/src/main/java/org/wso2/broker/client/http/HttpClient.java

```diff
@@ -0,0 +1,175 @@
+
+    private String encodedCredentials;
+
+    public HttpClient(Configuration configuration) {
+        this.url = configuration.getUrl() + BROKER_CONNECTION_URL_SUFFIX;
+        this.encodedCredentials = configuration.getEncodedCredentials();
+
+        // turn off SSL verification
+        try {
+            // Create a trust manager that does not validate certificate chains
+            TrustManager[] trustAllCerts = new TrustManager[] { new X509UntrustManagerImpl() };
+
+            // Install the all-trusting trust manager
+            SSLContext sc = SSLContext.getInstance("SSL");
+            sc.init(null, trustAllCerts, new java.security.SecureRandom());
+            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+
+            // Create all-trusting host name verifier
+            HostnameVerifier allHostsValid = (hostname, session) -> true;
+
+            // Install the all-trusting host verifier
+            HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
+        } catch (KeyManagementException | NoSuchAlgorithmException e) {
+            BrokerClientException exception = new BrokerClientException();
+            exception.addMessage("error initializing the connection");
+            exception.addMessage(e.getMessage());
+            throw exception;
+        }
+    }
+
+    // Send HTTP request
+    public HttpResponse sendHttpRequest(HttpRequest request, String httpMethod) {
+
+        try {
+
+            URL obj = new URL(url + request.getSuffix());
+            HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
+
+            //add request header
+            con.setRequestMethod(httpMethod);
+            con.setRequestProperty("User-Agent", USER_AGENT);
+            con.setRequestProperty("Accept", "application/json");
+            con.setRequestProperty("Content-Type", "application/json");
+            con.setRequestProperty("Authorization", "Basic" + encodedCredentials);
+
+            // set http data
+            if (Objects.nonNull(request.getQueryParameters()) || Objects.nonNull(request.getPayload())) {
+                con.setDoOutput(true);
+                DataOutputStream wr = null;
+                try {
+                    wr = new DataOutputStream(con.getOutputStream());
+                    if (Objects.nonNull(request.getQueryParameters())) {
+                        wr.writeBytes(request.getQueryParameters());
+                    }
+                    if (Objects.nonNull(request.getPayload())) {
+                        wr.writeBytes(request.getPayload());
+                    }
+                } finally {
+                    if (Objects.nonNull(wr)) {
+                        wr.flush();
+                        wr.close();
+                    }
+                }
+            }
+
+            int responseCode = con.getResponseCode();
+
+            BufferedReader in = null;
+            StringBuffer response = new StringBuffer();
```

Comment:
Shall we use StringBuilder here? StringBuilder is preferred when the object is accessed by a single thread. StringBuffer is thread safe hence less performant than StringBuilder due to synchronizations.

Reply:
+1